cmd/agent/app/proxy_builders.go

```diff
@@ -15,12 +15,18 @@
 package app
 
 import (
+	"google.golang.org/grpc/metadata"
+
 	"github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc"
 )
 
 // GRPCCollectorProxyBuilder creates CollectorProxyBuilder for GRPC reporter
 func GRPCCollectorProxyBuilder(builder *grpc.ConnBuilder) CollectorProxyBuilder {
 	return func(opts ProxyBuilderOptions) (proxy CollectorProxy, err error) {
-		return grpc.NewCollectorProxy(builder, opts.AgentTags, opts.Metrics, opts.Logger)
+		md := metadata.New(map[string]string{})
+		if builder.CollectorTenancyHeader != "" {
+			md = metadata.New(map[string]string{builder.CollectorTenancyHeader: builder.CollectorTenant})
```

Comment:
With this change every span received by the Agent will be tagged with a single tenant when forwarded to Collector. While this is a viable deployment model in some situations, I don't think it's ultimately the right one, especially when the Agent runs as a host agent receiving data from potentially multiple co-located applications (which could be different tenants).

For example, this makes it impossible to test multi-tenancy with all-in-one if the other tenants are also going through the agent.

My preference would be for the agent to extract the tenant directly from the spans it receives, and the current solution could be an opt-in extension to either override the tenant or inject it when empty (for cases when agent is used as a sidecar).